### ISSUE TITLE
Compile error when using square brackets inside inline footnote

### ISSUE THREAD

**Issue by nopria** [Witiko/markdown#453]
The following MWE does NOT compile:

```latex
\documentclass{article}
\usepackage[hybrid, inlineNotes=true,eagerCache=true]{markdown}

\begin{document}
\begin{markdown}

# MWE square brackets in inline footnotes with markdown:

Example something outside inline footnote^[Example square backets inside [...] inline footnote.].

\end{markdown}
\end{document}
```

Similarly to recently fixed bug #447 it seems that inline footnote is not recognized due to the presence of square brackets inside it.

The above axample compiles regularly with markdown version 2.23.0.

**Comment by Witiko** [comment]
Hello again and thanks for the bug report. It appears that the (inline) notes still have plenty edge cases to fix.

**Comment by Witiko** [comment]
The issue should be fixed in Markdown 3.7.0, to be released by the end of the month. Instructions for manual installation are the same as I described in https://github.com/Witiko/markdown/issues/447#issuecomment-2171781841. Sorry again and please, let me know if you spotted any more issues.

**Comment by nopria** [comment]
As soon as 3.7.0 is released I will test it on my big document which spotted the previous 2 bugs. Thanks you for fixing them.
